Jellyfin.Api/Controllers/SubtitleController.cs

```diff
@@ -0,0 +1,344 @@
+
+            if (item == null)
+            {
+                return NotFound();
+            }
+
+            _subtitleManager.DeleteSubtitles(item, index);
+            return NoContent();
+        }
+
+        /// <summary>
+        /// Search remote subtitles.
+        /// </summary>
+        /// <param name="id">The item id.</param>
+        /// <param name="language">The language of the subtitles.</param>
+        /// <param name="isPerfectMatch">Optional. Only show subtitles which are a perfect match.</param>
+        /// <response code="200">Subtitles retrieved.</response>
+        /// <returns>An array of <see cref="RemoteSubtitleInfo"/>.</returns>
+        [HttpGet("/Items/{id}/RemoteSearch/Subtitles/{language}")]
+        [Authorize]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        public async Task<ActionResult<RemoteSubtitleInfo[]>> SearchRemoteSubtitles(
+            [FromRoute] Guid id,
+            [FromRoute] string language,
+            [FromQuery] bool isPerfectMatch)
+        {
+            var video = (Video)_libraryManager.GetItemById(id);
+
+            return await _subtitleManager.SearchSubtitles(video, language, isPerfectMatch, CancellationToken.None).ConfigureAwait(false);
+        }
+
+        /// <summary>
+        /// Downloads a remote subtitle.
+        /// </summary>
+        /// <param name="id">The item id.</param>
+        /// <param name="subtitleId">The subtitle id.</param>
+        /// <response code="204">Subtitle downloaded.</response>
+        /// <returns>A <see cref="NoContentResult"/>.</returns>
+        [HttpPost("/Items/{id}/RemoteSearch/Subtitles/{subtitleId}")]
+        [Authorize]
+        [ProducesResponseType(StatusCodes.Status204NoContent)]
+        public ActionResult DownloadRemoteSubtitles(
+            [FromRoute] Guid id,
+            [FromRoute] string subtitleId)
+        {
+            var video = (Video)_libraryManager.GetItemById(id);
+
+            Task.Run(async () =>
+            {
+                try
+                {
+                    await _subtitleManager.DownloadSubtitles(video, subtitleId, CancellationToken.None)
+                        .ConfigureAwait(false);
+
+                    _providerManager.QueueRefresh(video.Id, new MetadataRefreshOptions(new DirectoryService(_fileSystem)), RefreshPriority.High);
+                }
+                catch (Exception ex)
+                {
+                    _logger.LogError(ex, "Error downloading subtitles");
+                }
+            });
+            return NoContent();
+        }
+
+        /// <summary>
+        /// Gets the remote subtitles.
+        /// </summary>
+        /// <param name="id">The item id.</param>
+        /// <response code="200">File returned.</response>
+        /// <returns>A <see cref="FileStreamResult"/> with the subtitle file.</returns>
+        [HttpGet("/Providers/Subtitles/Subtitles/{id}")]
+        [Authorize]
+        [ProducesResponseType(StatusCodes.Status200OK)]
```

Comment:
```suggestion
        [ProducesResponseType(StatusCodes.Status200OK)]
        [Produces(MediaTypeNames.Application.Octet)]
```

it may not actually return `application/octet`, but we should still specify so the requester knows it is receiving a stream